ci-operator/step-registry/ipi/conf/aws/route53/ipi-conf-aws-route53-commands.sh

```diff
@@ -0,0 +1,51 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+trap 'CHILDREN=$(jobs -p); if test -n "${CHILDREN}"; then kill ${CHILDREN} && wait; fi' TERM
+
+curl -L https://github.com/mikefarah/yq/releases/download/3.3.0/yq_linux_amd64 -o /tmp/yq && chmod +x /tmp/yq
+
+CONFIG="${SHARED_DIR}/install-config.yaml"
+
+
+# subnet, AZs and hosted zone
+allsubnetids="${SHARED_DIR}/allsubnetids"
+hostedzoneid="${SHARED_DIR}/hostedzoneid"
+availabilityzones="${SHARED_DIR}/availabilityzones"
+if [ ! -f "${allsubnetids}" ] || [ ! -f "${availabilityzones}" ] || [ ! -f "${hostedzoneid}" ]; then
+    echo "File ${allsubnetids} or ${availabilityzones} or ${hostedzoneid} does not exist."
+    exit 1
+fi
+
+echo -e "subnets: $(cat ${allsubnetids})"
+echo -e "AZs: $(cat ${availabilityzones})"
+echo -e "hosted zone: $(cat ${hostedzoneid})"
+
+CONFIG_ROUTE53_HOSTEDZONE="${SHARED_DIR}/install-config-route53.yaml.patch"
+cat > "${CONFIG_ROUTE53_HOSTEDZONE}" << EOF
+publish: External
+platform:
+  aws:
+    hostedZone: $(cat "${hostedzoneid}")
+    subnets: $(cat "${allsubnetids}")
+controlPlane:
+  platform:
+    aws:
+      zones: $(cat "${availabilityzones}")
+compute:
+- platform:
+    aws:
+      zones: $(cat "${availabilityzones}")
+EOF
```

Comment:
`publish` will be controlled by system environment in  `*-provision` chain, like https://github.com/openshift/release/pull/24888/files#diff-55d9cca8aaec70f1288e64a0af48754c084edcd57938c5b53ade1701ae6c4974R22-R24, so it should be removed from here.

there are some conflicts with https://github.com/openshift/release/pull/24888, the subnet and AZ should be set in `ipi-conf-aws-custom-vpc`, which is newly introduced in https://github.com/openshift/release/pull/24888, but it has not been merged yet, hope it can be merged within this week.

eventually, this part should be:
```
cat > "${CONFIG_ROUTE53_HOSTEDZONE}" << EOF
platform:
  aws:
    hostedZone: $(cat "${hostedzoneid}")
EOF
```

Reply:
Sure, I will update once this https://github.com/openshift/release/pull/24888 merged to the master.